client/windows/CCastleInterface.cpp

```diff
@@ -990,7 +996,8 @@ void CCastleInterface::recreateIcons()
 
 	hall = new CTownInfo( 80, 413, town, true);
 	fort = new CTownInfo(122, 413, town, false);
-
+	fastArmyPurhase = std::make_shared<CButton>(Point(122, 413), "itmcl.def", CButton::tooltip("quick army purhase"), [&](){builds->enterToTheQuickRecruitmentWindow();});

```

Comment:
Please do not use smart pointers to `CIntObject` descendants when DISPOSE capture flag set (OBJ_CONSTRUCTION_CAPTURING_ALL macro set all flags)